syntax = "proto3";

option java_multiple_files = true;
option java_package = "productService.gRPC";
option java_outer_classname = "alternative";

package productService;

message OrderItem {
  string product_id = 1;
  int32 quantity = 2;
  float price = 3;
}

message OrderMessage {
  string customer_id = 1;
  repeated OrderItem items = 2;
}

message OrderConfirmation {
    string confirmation_id = 1;
    string message = 2;
}

message Product {
  int32 id = 1;
  string name = 2;
  float price = 3;
  string description = 4;
  string image = 5;
}

message ProductList{
  repeated Product products = 1;
}

message ProductRequest {
  int32 id = 1;
}

message Empty {}

service OrderService {
    rpc SubmitOrder (OrderMessage) returns (OrderConfirmation);
}

service ProductService {
    rpc GetProducts(Empty) returns (ProductList);
    rpc GetProductById(ProductRequest) returns (Product);
}